### PR TITLE
PLAT-750: Set Memory after delegation and ordered call

### PR DESCRIPTION
**Pull request by ElSacramento**
update memory cache after tolerable call and delegated finished

### CODE REVIEW

**Review by MariaVasilenko — DISMISSED** [review]

File: ledger-core/virtual/handlers/vdelegatedrequestfinished.go [hunk 1]
```diff
@@ -134,6 +140,7 @@ func (s *SMVDelegatedRequestFinished) stepProcess(ctx smachine.ExecutionContext)
 		}
 
 		s.updateSharedState(ctx, state)
+		s.updateMemoryCache(ctx, state)
```
Comment (bigbes) [inline ledger-core/virtual/handlers/vdelegatedrequestfinished.go:143]: Ehm, you make async call request from there. Are you really need to do it from that function?

File: ledger-core/ledger/server/lmnapp/lmn_app.go [hunk 2]
```diff
@@ -39,6 +39,7 @@ func NewAppCompartment(_ configuration.Ledger, comps insapp.AppComponents) *insc
 		func(ctx context.Context, _ injector.DependencyInjector, setup insconveyor.AppCompartmentSetup) insconveyor.AppCompartmentSetup {
 
 			setup.Dependencies.AddInterfaceDependency(&comps.MessageSender)
+			setup.Dependencies.AddInterfaceDependency(&comps.MemoryCache)
```
Comment (aalekseev-ru) [inline ledger-core/ledger/server/lmnapp/lmn_app.go:42]: do we need it here?

**Review by aalekseev-ru — DISMISSED** [review]

File: ledger-core/ledger/server/lmnapp/lmn_app.go (hunk 2 above)
Comment (ElSacramento) [reply ledger-core/ledger/server/lmnapp/lmn_app.go]: removed

File: ledger-core/virtual/handlers/vdelegatedrequestfinished.go (hunk 1 above)
Comment (ElSacramento) [reply ledger-core/virtual/handlers/vdelegatedrequestfinished.go]: fixed

**Review by aalekseev-ru — APPROVED** [review]

**Review by bigbes — APPROVED** [review]

**Pull request merged**